apps/script/build-color-js.js

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env node
+// Script to generate color.js from color.scss
+var fs = require('fs');
+var path = require('path');
+var readline = require('readline');
+
+var colorScssPath = path.resolve('../shared/css/color.scss');
+var colorJsPath = path.resolve('./src/color.js');
+
+// Regular expression to capture a color definition from SCSS
+var colorRe = /\$(\w+)\s*:\s*([^;]+);/;
+var cachedColors = {};
+
+// In color.scss some color definitions reference previous
+// color definitions.  Since we're reading them in order,
+// we can recursively resolve these referential defintions
+// until we land on an actual color.
+function resolveColor(value) {
+  var originalValue = value;
+  while (/^\$/.test(value)) {
+    value = cachedColors[value.substring(1)];
+  }
+  return value;
+}
+
+// Generate color.js while reading color.scss line-by-line
+var out = fs.createWriteStream(colorJsPath);
+out.write([
+  '// color.js',
+  '// GENERATED FILE: DO NOT MODIFY DIRECTLY',
+  '// This generated file exports all colors defined in color.scss',
+  '// for use in JavaScript. The generator script is build-color-js.js',
+  'module.exports = {\n'
+].join('\n'));
+
+var rl = readline.createInterface({
+  input: fs.createReadStream(colorScssPath),
+  terminal: false
+});
+
+var currentLine = 0;
+rl.on('line', function (line) {
+  currentLine++;
+  var match = colorRe.exec(line);
+  if (match === null) {
+    return;
+  }
+
+  var colorName = match[1];
+  var colorValue = resolveColor(match[2]);
+  if (typeof colorValue === 'undefined') {
+    throw new Error([
+        'Unable to resolve color ' + colorName,
+        colorScssPath + ':' + currentLine,
+        line,
+        ' ^'
+        ].join('\n'));
```

Comment:
I did my best here to mimic standard node error messages, which makes this stand out nicely when something goes wrong.  Here's an example failure running `npm run build` when there's a bad reference in color.scss:

```
Running "exec:buildColorJs" (exec) task
>> /home/brad/Projects/code-dot-org/apps/script/build-color-js.js:52
>>     throw new Error([
>>           ^
>> Error: Unable to resolve color broken_color
>> /home/brad/Projects/code-dot-org/shared/css/color.scss:10
>> $broken_color: $teal;
>>  ^
>>     at Interface.<anonymous> (/home/brad/Projects/code-dot-org/apps/script/build-color-js.js:52:11)
>>     at Interface.emit (events.js:107:17)
>>     at Interface._onLine (readline.js:214:10)
>>     at Interface.<anonymous> (readline.js:344:12)
>>     at Array.forEach (native)
>>     at Interface._normalWrite (readline.js:343:11)
>>     at ReadStream.ondata (readline.js:91:10)
>>     at ReadStream.emit (events.js:107:17)
>>     at readableAddChunk (_stream_readable.js:163:16)
>>     at ReadStream.Readable.push (_stream_readable.js:126:10)
>> Exited with code: 1.
Warning: Task "exec:buildColorJs" failed. Use --force to continue.
```